### ISSUE TITLE
facets' Hash#except! conflicts with rails/activesupport implementation

### ISSUE THREAD

**Issue by fsabe** [rubyworks/facets#248]
Rails (ativerecord/activesupport) expects Hash#except! to return the hash itself without the removed keys:

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/has_many_through_association.rb#L79
```
        def through_scope_attributes
          scope.where_values_hash(through_association.reflection.name.to_s).
            except!(through_association.reflection.foreign_key,
                    through_association.reflection.klass.inheritance_column)
        end
```

https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/hash/except.rb#L18
```
  def except!(*keys)
    require "pry"; binding.pry
    keys.each { |key| delete(key) }
    self
  end
```

But https://github.com/rubyworks/facets/blob/master/lib/core/facets/hash/except.rb implements a different behavior and returns the removed keys instead:
 
```
  def except!(*rejected)
    removed = {}
    rejected.each{ |k| removed[k] = delete(k) }
    removed
  end
```

This conflict causes errors when activerecord is used alongside facets.

Tested using the latest activerecord version 4.2.1, activesupport 4.2.1 and facets 2.9.3.

**Comment by fsabe** [comment]
@leocwolter

**Comment by bigtunacan** [comment]
+1

**Comment by trans** [comment]
Yes, you are right. I did some checking and all Ruby bang methods seem to work the way Rails' `except!` does. That's too bad actually b/c it is not as useful as what I did with Facets. With Facets, it is easy to see what entires were *actually* removed. To do the same the other way means first fully duplicating the hash and then "subtracting" the old hash from it. And, LOL, I don't think Ruby or Rails has a method to do that. Thankfully Facets does: `Hash#-`.

So, in any case, I will change it. But I added a [feature request](https://bugs.ruby-lang.org/issues/12360) for Ruby so maybe in the future we can have more useful bang method return values.
